docs/proposals/downsampling.md

```diff
@@ -0,0 +1,161 @@
+downsampling has to take into account dependencies concerning other 
+tenant-related work streams, including but not limited to
+[Deletion of Tenant Data from Blocks Storage][tenant-deletion],
+[Retention of Tenant Data from Blocks Storage][tenant-retention], and
+[Parallel Compaction by Time Interval][parallel-compaction].
+
+### Prior Art
+
+Cortex is not the first Prometheus LTS solution offering downsampling. For
+example, [Thanos][thanos-ds] offers a comprehensive downsampling as part of
+the compactor, [M3][m3-ds] supports trimmed downsampling via the aggregator,
+and [InfluxDB][influxdb-ds] allows users to perform downsampling via a
+continuous query.
+
+## Potential Solutions
+
+Based on the considerations discussed above as well as the review of prior art
+in other Prometheus ecosystem projects, there are two principled
+solution approaches for Cortex one can think of. Before we get into the details
+let us step back and review the write path:
+
+```
+
+            │
+        [1] │
+            │
+            │
+            ▼
+   ┌────────────────┐     ┌────────────────┐
+   │   distributor  │     │   distributor  │
+   └────┬───────────┘     └────────────────┘
+        │
+        │
+   [1]  │
+        │
+        ▼
+ ┌────────────┐  ┌────────────┐  ┌────────────┐
+ │  ingester  │  │  ingester  │  │  ingester  │
+ └─────┬──────┘  └────────────┘  └────────────┘
+       │
+       │
+       │
+  [1]  │
+       │
+       │         ┌────────────┐
+       │         │            │     [2]   ┌─────────────┐
+       └───────► │   object   │ ◄─────────┤  compactor  │
+                 │   store    │           └─────────────┘
+                 │            │
+                 └────────────┘
+```
+
+We have two ways to manipulate the blocks and indices: [1] is during the
+ingestion of the samples, and [2] represents the offline case, operating on
+already ingested data.
+
+### Offline
+
+One solution would be implementing downsampling in Cortex as part of the
+compactor [2]. This would result in a design that is, at least on a high level,
+aligned with Thanos and would allow both comprehensive as well as 
+trimmed downsampling.
+
+### Streaming
+
+If one were to implement downsampling in Cortex along the write path as
+hinted at in [1], either in the distributor or ingester, only trimmed
+downsampling would be supported.
+
+## Proposed Solution
+
+We now turn our attention to the proposed solution for downsampling in Cortex.
+
+### Requirements
+
+* MUST support opt-in (default behavior is as without downsampling).
+* MUST support tenant-level granularity (can opt-in on an per-tenant basis).
+* SHOULD not have a significant impact on the write path performance.
+
+Concerning the sampling tiers we want to support the following time periods:
+
+1. Data up to one week raw (not downsampled).
+1. Data older than one week up to six months downsampled to 1h intervals.
+1. Data older than six months downsampled to 24h intervals.
```

Comment:
Would this be UTC or configurable per tenant? Does timezone ever get considered in a 24h downsample?

Reply:
Thanks for the question! Yes, my expectation was per-tenant and will make sure to address the time zone question in the next iteration of the design.